src/decoder/mod.rs

```diff
@@ -475,11 +475,21 @@ fn fix_endianness_and_predict(
     }
 }
 
-fn invert_colors(buf: &mut [u8], color_type: ColorType, sample_format: SampleFormat) {
+fn invert_colors(
+    buf: &mut [u8],
+    color_type: ColorType,
+    sample_format: SampleFormat,
+) -> TiffResult<()> {
     match (color_type, sample_format) {
-        (ColorType::Gray(8), SampleFormat::Uint) => {
+        (ColorType::Gray(n), SampleFormat::Uint) if 8 % n == 0 => {
```

Comment:
This condition might be wrong. It doesn't cover `n=1`, but does overlap with the n=16, 32, and 64 cases below...

Though actually I think the `for x in buf { *x = !*x; }` loop should work for any `SampleFormat::Uint` value

Reply:
Quite the other way around, it covers n=1,2,4,8

Reply:
```suggestion
        (ColorType::Gray(n), SampleFormat::Uint) if 8 % n == 0 => {
        (ColorType::Gray(1 | 2 | 4 | 8), SampleFormat::Uint) => {
```

This spells out the values explicitly

Reply:
For some reason I thought or-patterns beyond the top-level were quite recent. But it's 1.56. Is this grey beard status? But don't want any beard, let alone grey :crying_cat_face: